src/dotnet/commands/dotnet-install/InstallCommand.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.DotNet.ShellShimMaker;
+using Microsoft.DotNet.ToolPackageObtainer;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using NuGet.Build.Tasks;
+using NuGet.Configuration;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class InstallCommand
+    {
+        public static int Run(string[] args)
+        {
+            CommandLine.Parser parser = Parser.Instance;
+            ParseResult result = parser.ParseFrom("dotnet install", args);
+
+            result.ShowHelpOrErrorIfAppropriate();
+
+            AppliedOption parseResult = result["dotnet"]["install"]["tool"];
+
+            var packageId = parseResult.Arguments.Single();
+            var packageVersion = parseResult.ValueOrDefault<string>("version");
+
+            FilePath? configFile = null;
+
+            var configFilePath = parseResult.ValueOrDefault<string>("configfile");
+
+            if (configFilePath != null)
+            {
+                configFile = new FilePath(configFilePath);
```

Comment:
Should we do validation that the file actually exists? Or we will just rely on NuGet for that? You should at least try it out and see that it is a good experience when that happens.

Reply:
In short I'll add validation.

There are several layer to this:

In general. We are using NuGet underneath is not an implementation detail. For example, the user need to set up local **NuGet feed**, if they want to use locally. As a result of NuGet no being an implementation detail, the existence of the config file is not enough, it has to be xml and it has to be in a special form NuGet defined.

So I would let NuGet to handle the validation

However, there is one step before that. We need to copy that file to temp directory as walk around for no supporting of configfile in `nuget.xplt package add`. This is an implementation detail. When it throws, it should have meaning full message.

Here is the current error message:
```
C:\Users\wul>dotnet install tool demo.hello --configfile C:\Users\Desktop\NuGet.Config.txt
System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\Desktop\NuGet.Config.txt'.
   at System.IO.Win32FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite)
   at System.IO.File.InternalCopy(String sourceFileName, String destFileName, Boolean overwrite)
   at System.IO.File.Copy(String sourceFileName, String destFileName)
   at Microsoft.DotNet.ToolPackageObtainer.ToolPackageObtainer.InvokeAddPackageRestore(Nullable`1 nugetconfig, FilePath tempProjectPath, String packageId)
   at Microsoft.DotNet.ToolPackageObtainer.ToolPackageObtainer.ObtainAndReturnExecutablePath(String packageId, String packageVersion, Nullable`1 nugetconfig, String targetframework)
   at Microsoft.DotNet.Cli.InstallCommand.ObtainPackage(String packageId, String packageVersion, Nullable`1 configFile, String framework, DirectoryPath executablePackagePath)
   at Microsoft.DotNet.Cli.InstallCommand.Run(String[] args)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
```